java/client/src/main/java/glide/api/models/ClusterValue.java

```diff
@@ -0,0 +1,44 @@
+package glide.api.models;
+
+import java.util.Map;
+
+/**
+ * A union-like type which can store single or bulk value retrieved from Redis.
+ *
+ * @param <T> The wrapped data type
+ */
+public class ClusterValue<T> {
+    private Map<String, T> multiValue = null;
+
+    private T singleValue = null;
+
+    private ClusterValue() {}
+
+    /** Get per-node value. */
```

Comment:
We will need a more explicit javadoc if we're going to expose this to the user. 

Please mention that hasMultiValue() should be called first to avoid exceptions